tailscale/data_source_user.go

```diff
@@ -78,32 +72,65 @@ func dataSourceUser() *schema.Resource {
 	return &schema.Resource{
 		Description: "The user data source describes a single user in a tailnet",
 		ReadContext: dataSourceUserRead,
-		Schema:      userSchema,
+		Schema: combinedSchemas(commonUserSchema, map[string]*schema.Schema{
+			"login_name": {
+				Type:        schema.TypeString,
+				Description: "The emailish login name of the user.",
+				Optional:    true,
```

Comment:
Now that this is the only way to query a user:
```suggestion
				Required:    true,
```

Reply:
It occurred to me that I'm still allowing use of ID, so I've updated the schema to make this clear.